pkg/console/operator/status.go

```diff
@@ -0,0 +1,295 @@
+package operator
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+// Operator status is set on the Operator Config, which is:
+//   group: console.operator.openshift.io
+//   kind:  Console
+//   name:  console
+// And is replicated out onto the the clusteroperator by a separate sync loop:
+//   group: config.openshift.io
+//   kind:  ClusterOperator
+//   name:  console
+//
+// Status Condition Types
+// Status conditions should not be set to a "default" state when the operator starts up.
+// Instead, set a status explicitly only when the state is known.  This is because
+// the various status conditions may have been set on previous runs of the sync loop, or
+// possibly even by a previous operator container.
+//
+// Available = the operand (not the operator) is available
+//    example: the console resources are stabilized.
+//    example: the console is not yet in a functional state, one or
+//      more resources may be missing.
+// Progressing = the operator is trying to transition operand state
+//    example: during the initial deployment, the operator needs to create a number
+//      of resources.  the console (operand) is likely in flux.
+//    example: the assigned route .spec.host changes for the console.  the oauthclient
+//      must be updated, as must the configmap, etc.  numerous resources are in flux,
+//      the operator reports progressing until the resources stabilize
+// Failing = the operator (not the operand) is failing
+//    example: The console operator is unable to update the console config with
+//      a new logoutRedirect URL. The operator is failing to do its job, however the
+//      console (operand) may or may not be functional (see Available above).
+//
+//
+// Status Condition Reason & Message
+// Reason:  OperatorSyncLoopError
+// Message: "The operator sync loop was not completed successfully"
+//
+//
+
+// Lets transition to using this, and get the repetition out of all of the above.
+func (c *consoleOperator) SyncStatus(operatorConfig *operatorsv1.Console) (*operatorsv1.Console, error) {
+	logConditions(operatorConfig.Status.Conditions)
+	updatedConfig, err := c.operatorConfigClient.UpdateStatus(operatorConfig)
+	if err != nil {
+		return nil, fmt.Errorf("status update error: %v \n", err)
```

Comment:
It looks like these errors will never get logged if `SyncStatus` fails

Reply:
The first thing this func does is call `logConditions`, which will output all the conditions in the logs. Then it will try to make an API call, and error out if something fails.

Reply:
Right, but that error is never logged. We return an error from this function, but none of the callers handle it. So we'd never know that status isn't updated.